content/articles/tracking-of-objects-using-the-histogram-based-tracker-in-matlab/index.md

```diff
@@ -0,0 +1,184 @@
+
+The computer recognizes three colour channels. They are red, green and blue. If you use an integer of 1, it represents the red colour channel. The integer 2 represents green colour channels, and finally, 3 represents blue. The colour channel you choose to use depends on the object being tracked and the surrounding environment.
+
+In our case, we use a green channel since the ball color is close to the green channel than the red or blue.
+To apply tracking on our subsequent video frames, we use a `while` loop.
+
+```matlab
+%% Track object
+idx = 1;
+while ~isDone(videoFReader)
+    img = step(videoFReader);
+    imgYcbcr = rgb2ycbcr(img);
+    [bbox,~,score(idx)] = step(tracker,imgYcbcr(:,:,2));
+    out = insertShape(img,'Rectangle',bbox);
+    step(vidPlayer,out);
+    idx = idx + 1;
+    pause(0.1)
+end
+```
+
+The loop first checks if there is a video frame using the `~isDone()` function. It then reads the video frame using the `step()` function. This video frame is then converted from RGB colour space to ycbcr colour space in the loop. This conversion is done using `rgb2ycbcr()` function. 
+
+We use `step` on the histogram-based tracker object with the green channel of the current video frame to track the object. We need to have a bounding box around the object of interest. We use the `insertShape()` function to insert a bounding box. This function takes in the image and the shape of the required shape of the bounding box.
+
+Finally, we display the output using the `step()` function. This function takes the video player `vidPlayer` and the image `out`. This image is a video frame, but the object of interest is bounded using a `rectangular` bounding box.
+
+A video is a combination of frames that keep changing. Since this is a combination, the while loop ensures that all the video frames are operated. It also ensures the object of interest is bounded. To visualize the output, add a pause of `0.1` seconds. 
+
+The index `idx` is used to plot the confidence score. The confidence score is used to see the accuracy of the tracking process. When the confidence score is high, the accuracy of the tracking process is likely to be high. When it is low, then the opposite applies.
+
+Now, let us plot the confidence score:
+
+```matlab 
+figure;
+plot(score)
+xlabel('Frame #')
+ylabel('Confidence Score (0,1)')
+```
+
+From the code above, we created a `figure` in which the plotting should occur. We then used the `plot()` function to plot these scores. The `xlabel` is given as `frame` and the `ylabel` as the confidence score. The y-axis should move up to a maximum of 1.
+
+Once all the tracking process is done, clean the video player `vidPlayer` and the video frame reader `videoFReader`. It is done using the `release()` function as shown below:
+
+```Matlab
+%% Clean up
+release(vidPlayer);
+release(videoFReader);
+```
+
+If we run this code at this point, we get the following output:
+
+![Tracking](tracking-one.png)
```

Comment:
kindly link your images using the correct format:
`/engineering-education/folder-name/image-name`